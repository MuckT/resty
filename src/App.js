import React, { useState } from 'react';
import './App.scss';
import Footer from './Components/Footer/Footer';
import RequestForm from './Components/Form/Form';
import Header from './Components/Header/Header';
import Response from './Components/Results/Results';
import History from './Components/History/History';

const axios = require('axios');

function App() {
  const [appState, setAppState] = useState(
    {
      'url': '',
      'method': 'GET',
      'results': null,
      'status': null,
      'history': []
    }
  )

  const [isLoading, setLoading] = useState(false)

  const handleSubmit = async (method, url, body = {}) => {
    try {
      setLoading(true)
      let response = await axios({
        method: method,
        url: url,
        data: body,
        validateStatus: function (status) {
          return status;
        }
      })
      // Create result to use for history
      const result = {
        'method': method,
        'url': url,
        'results': response.data,
        'status': response.status
      }
      setAppState(
        {
          'method': method,
          'url': url,
          'results': response.data,
          'status': response.status,
          'history': [...appState.history, result]
        }
      )
    } catch (error) {
      console.warn(error)
    } finally {
      setLoading(false)
    }
  }
  
  const historyClick = (historyItem) => {
    setAppState(
      {
        'method': historyItem.method,
        'url': historyItem.url,
        'results': historyItem.results,
        'status': historyItem.status,
        'history': appState.history
      }
    )
  }

  return (
    <div className="App">
      <Header />
      <div className="main">
        <RequestForm handleSubmit={handleSubmit}/>
        <Response status={appState.status} results={appState.results} isLoading={isLoading}/>
        {appState.history.length ? <History historyClick={historyClick} history={appState.history}></History> : null}
      </div>
      <Footer />
    </div>
  );
}

export default App;