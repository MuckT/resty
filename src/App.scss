@import "custom";

.App {
  height: 100vh;
  text-align: center;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr auto;
  grid-template-areas:
    'header'
    'main'
    'footer'
}

.App header {
  grid-area: header;
}

.App main {
  grid-area: main;
}

.App footer{
  grid-area: footer;
}
