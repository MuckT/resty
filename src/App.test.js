import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import App from './App';
import RequestForm from './Components/Form/Form'

it('renders RESTy header', () => {
  render(<App />);
  const linkElement = screen.getByText("RESTy");
  expect(linkElement).toBeInTheDocument();
});

it('should invoke handleSubmit with correct parameters', () => {
  const mockFn = jest.fn();
  render(<RequestForm handleSubmit={mockFn}/>);
  const input = screen.getByRole('textbox') 
  let get = screen.getByRole('button', {
    name: /get/i
  })
  let submit = screen.getByRole('button', {
    name: /submit/i
  })
  fireEvent.change(input, {target: {value: 'https://pokeapi.co/api/v2/'}})
  fireEvent.click(get)
  fireEvent.click(submit)
  expect(mockFn).toBeCalledWith("GET", "https://pokeapi.co/api/v2/");
})

it('should match snapshot', () => {
  expect(render(<App />)).toMatchSnapshot()
})
