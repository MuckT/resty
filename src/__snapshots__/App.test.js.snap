// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <h1>
            RESTy
          </h1>
        </header>
        <div
          class="main"
        >
          <form
            class=""
          >
            <div
              class="m-3"
            >
              <div
                class="input-group"
              >
                <span
                  class="input-group-text"
                >
                  URL
                </span>
                <input
                  class="form-control"
                  id="testFormUrlInput"
                  placeholder="https://example.com"
                  required=""
                  type="url"
                  value=""
                />
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  Submit
                </button>
              </div>
            </div>
            <div
              class="mb-2 btn-group-vertical"
              role="group"
            >
              <input
                autocomplete="off"
                checked=""
                class="btn-check"
                id="radio-0"
                name="radio"
                type="radio"
                value="GET"
              />
              <label
                class="btn btn-outline-primary"
                for="radio-0"
                role="button"
                tabindex="0"
              >
                GET
              </label>
              <input
                autocomplete="off"
                class="btn-check"
                id="radio-1"
                name="radio"
                type="radio"
                value="POST"
              />
              <label
                class="btn btn-outline-primary"
                for="radio-1"
                role="button"
                tabindex="0"
              >
                POST
              </label>
              <input
                autocomplete="off"
                class="btn-check"
                id="radio-2"
                name="radio"
                type="radio"
                value="PUT"
              />
              <label
                class="btn btn-outline-primary"
                for="radio-2"
                role="button"
                tabindex="0"
              >
                PUT
              </label>
              <input
                autocomplete="off"
                class="btn-check"
                id="radio-3"
                name="radio"
                type="radio"
                value="PATCH"
              />
              <label
                class="btn btn-outline-primary"
                for="radio-3"
                role="button"
                tabindex="0"
              >
                PATCH
              </label>
              <input
                autocomplete="off"
                class="btn-check"
                id="radio-4"
                name="radio"
                type="radio"
                value="DELETE"
              />
              <label
                class="btn btn-outline-primary"
                for="radio-4"
                role="button"
                tabindex="0"
              >
                DELETE
              </label>
            </div>
          </form>
          <section>
            <pre
              id="results"
            />
          </section>
        </div>
        <footer
          id="footer"
        >
          © 2021
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <h1>
          RESTy
        </h1>
      </header>
      <div
        class="main"
      >
        <form
          class=""
        >
          <div
            class="m-3"
          >
            <div
              class="input-group"
            >
              <span
                class="input-group-text"
              >
                URL
              </span>
              <input
                class="form-control"
                id="testFormUrlInput"
                placeholder="https://example.com"
                required=""
                type="url"
                value=""
              />
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
          <div
            class="mb-2 btn-group-vertical"
            role="group"
          >
            <input
              autocomplete="off"
              checked=""
              class="btn-check"
              id="radio-0"
              name="radio"
              type="radio"
              value="GET"
            />
            <label
              class="btn btn-outline-primary"
              for="radio-0"
              role="button"
              tabindex="0"
            >
              GET
            </label>
            <input
              autocomplete="off"
              class="btn-check"
              id="radio-1"
              name="radio"
              type="radio"
              value="POST"
            />
            <label
              class="btn btn-outline-primary"
              for="radio-1"
              role="button"
              tabindex="0"
            >
              POST
            </label>
            <input
              autocomplete="off"
              class="btn-check"
              id="radio-2"
              name="radio"
              type="radio"
              value="PUT"
            />
            <label
              class="btn btn-outline-primary"
              for="radio-2"
              role="button"
              tabindex="0"
            >
              PUT
            </label>
            <input
              autocomplete="off"
              class="btn-check"
              id="radio-3"
              name="radio"
              type="radio"
              value="PATCH"
            />
            <label
              class="btn btn-outline-primary"
              for="radio-3"
              role="button"
              tabindex="0"
            >
              PATCH
            </label>
            <input
              autocomplete="off"
              class="btn-check"
              id="radio-4"
              name="radio"
              type="radio"
              value="DELETE"
            />
            <label
              class="btn btn-outline-primary"
              for="radio-4"
              role="button"
              tabindex="0"
            >
              DELETE
            </label>
          </div>
        </form>
        <section>
          <pre
            id="results"
          />
        </section>
      </div>
      <footer
        id="footer"
      >
        © 2021
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
